### ISSUE TITLE
YouTube: This format is DRM protected

### ISSUE THREAD

**Issue by Anuskuss** [yt-dlp/yt-dlp#7820]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

All of the sudden I can't download "YouTube Movies & TV" trailers in the highest quality anymore. I checked with `--allow-u` in case it's trying to download the movie instead (DRM protection would make sense then) but that wasn't the case.
`--check-formats` doesn't do anything.

These are off-limits:
```
ID  EXT   RESOLUTION FPS CH │   FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
290 mp4   1920x1080         │ ~ 66.45MiB 4536k m3u8  │ avc1.640029 4536k video only
291 mp4   1920x1080         │ ~ 88.84MiB 6065k m3u8  │ avc1.640029 6065k video only
292 mp4   1920x1080         │ ~121.83MiB 8317k m3u8  │ avc1.640029 8317k video only
```

These are okay:
```
ID  EXT   RESOLUTION FPS CH │   FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────
270 mp4   1920x1036   25    │ ~ 66.23MiB 4521k m3u8  │ avc1.640028 4521k video only
137 mp4   1920x1036   25    │   40.85MiB 2854k https │ avc1.640028 2854k video only          1080p, mp4_dash
216 mp4   1920x1036   25    │   75.87MiB 5300k https │ avc1.640028 5300k video only          1080p, mp4_dash
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.youtube.com/watch?v=oAX5munikYU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481]
[debug] Python 3.11.3 (CPython x86_64 64bit) - Linux-6.4.10-arch1-1-x86_64-with-glibc2.38 (OpenSSL 3.1.2 1 Aug 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, brotlicffi-1.0.9.2, certifi-2023.07.22, sqlite3-2.6.0, websockets-10.4, xattr-0.10.1
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[youtube] Extracting URL: https://www.youtube.com/watch?v=oAX5munikYU
[youtube] oAX5munikYU: Downloading webpage
WARNING: [youtube] unable to extract initial player response; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] oAX5munikYU: Downloading ios player API JSON
[youtube] oAX5munikYU: Downloading android player API JSON
[youtube] oAX5munikYU: Downloading iframe API JS
[youtube] oAX5munikYU: Downloading player 3cd2d050
[youtube] oAX5munikYU: Downloading web player API JSON
[youtube] Extracting URL: tnp_1rzDmro
[youtube] tnp_1rzDmro: Downloading webpage
WARNING: [youtube] unable to extract initial player response; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] tnp_1rzDmro: Downloading ios player API JSON
[youtube] tnp_1rzDmro: Downloading android player API JSON
[youtube] tnp_1rzDmro: Downloading iframe API JS
[youtube] tnp_1rzDmro: Downloading web player API JSON
[youtube] tnp_1rzDmro: Downloading m3u8 information
WARNING: [youtube] unable to extract yt initial data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
WARNING: [youtube] Incomplete data received in embedded initial data; re-fetching using API.
[youtube] tnp_1rzDmro: Downloading initial data API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] tnp_1rzDmro: Downloading 1 format(s): 292+140
[debug] Invoking hlsnative downloader on "https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1691898182/ei/5vzXZJbXBM2O1gLeyZ2QBg/ip/###/itag/292/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/gir/yes/clen/113512896/lmt/1439334338669851/dur/120.080/hls_chunk_host/rr2---sn-h0jeenld.googlevideo.com/mh/wf/mm/31,26/mn/sn-h0jeenld,sn-4g5edn6y/ms/au,onr/mv/m/mvi/2/pl/40/initcwndbps/2085000/vprv/1/playlist_type/DVR/dover/13/mt/1691876230/fvip/2/short_key/1/keepalive/yes/fexp/24007246/beids/24350017/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,gir,clen,lmt,dur,vprv,playlist_type/sig/AOq0QJ8wRQIgXvjKqeM9meyfxlstKJ7WBmbdlDiYfl2W4ANitIeVO6kCIQDtDoA4D_SGDJs7hjPXZdJFFnKTCOvUB2VaCgKRcpL9Kw%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AG3C_xAwRAIgDJ0f3JbdDGs6F7IWUs3v1zmjy8OxdHYPmaZC6qcoda0CIGaLa070LyNxkAJLWm_e6v61tqLHM7nVkpPmILOZlPkA/playlist/index.m3u8"
[hlsnative] Downloading m3u8 manifest
ERROR: This format is DRM protected; Try selecting another format with --format or add --check-formats to automatically fallback to the next best format

[debug] Invoking http downloader on "https://rr2---sn-h0jeenld.googlevideo.com/videoplayback?expire=1691898182&ei=5vzXZJbXBM2O1gLeyZ2QBg&ip=###&id=o-AOhUUp8Wl71t-hUUBbXEPnOdHyzta7Tbw29LqllJobyP&itag=140&source=youtube&requiressl=yes&mh=wf&mm=31%2C26&mn=sn-h0jeenld%2Csn-4g5edn6y&ms=au%2Conr&mv=m&mvi=2&pl=40&initcwndbps=2085000&vprv=1&svpuc=1&mime=audio%2Fmp4&gir=yes&clen=1942640&dur=120.140&lmt=1523998018350947&mt=1691876230&fvip=2&keepalive=yes&fexp=24007246&beids=24350017&c=IOS&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAKAy5SFKx4BeWNQWt_VtreJuwsqxiqZrmSzZDzNUXNXiAiBaF4A4yB8X-klMYOR2jikGsh6SdZ_2OZ116oGhq9iLfg%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhANCY-J8SKkPrS8qrM6TSaQIvbGSEJmARx5WOFzXRpOoIAiAdYWOi1ZahwqDU-3m5I4jeflfLb3t4sShZB5gVZGdqVw%3D%3D"
[download] Destination: Silver Linings - Trailer [tnp_1rzDmro].f140.m4a
[download] 100% of    1.85MiB in 00:00:00 at 7.27MiB/s
```

**Comment by bashonly** [comment]
>  Try selecting another format with --format or add --check-formats to automatically fallback to the next best format

Did you try another video format? It may just be the iOS m3u8 format that is DRM-protected. iOS formats have only been extracted by default since version 2023.06.22, which could explain why you hadn't experienced this issue before. (Also note that `--check-formats` is not working properly in 2023.07.06, but has been fixed in master)

I tried to reproduce this issue, but I think the OP's video may be geo-restricted to Germany:
```
[youtube] oAX5munikYU: Video unavailable. This video is not available
```

**Comment by Anuskuss** [comment]
You're right, I should've mentioned that I already tried `--check-formats` and that it didn't work, and also which formats are allowed and which are (apparently) DRM-protected. Updated the OP.

**Comment by october262** [comment]
is this the same thing  - https://www.youtube.com/watch?v=7jiytWGenaI
as this  - https://www.youtube.com/watch?v=oAX5munikYU ??

**Comment by Anuskuss** [comment]
No, this one only has
```
ID  EXT   RESOLUTION FPS CH │  FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
137 mp4   1920x1080   24    │  62.53MiB 3743k https │ avc1.640028 3743k video only          1080p, mp4_dash
```
which downloads fine (needs an US VPN and audio is English).

**Comment by pukkandan** [comment]
--check-formats is broken on last release. Pls try master/nightly, or use --check-all-formats as workaround

**Comment by pukkandan** [comment]
Fixed by https://github.com/yt-dlp/yt-dlp/commit/8cb7fc44db010e965d808ee679ef0725cb6e147c

**Comment by Anuskuss** [comment]
@pukkandan Okay but what about the 3 streams? Why are they DRM-protected? I would've thought that they might have been misdetected.

**Comment by pukkandan** [comment]
Some IOS formats are DRM protected even for otherwise unprotected video. Does nightly/master show "Maybe DRM" for these formats in `-F`? If you download them with `--allow-unplayable`, do they play? If it is labeled and do not play, nothing more for us to do. If not, pls attach `.dump`s of `--write-page -f290` along with corresponding verbose logs

**Comment by Anuskuss** [comment]
Latest nightly (2023.08.14.182926) does not say "Maybe DRM" but `--allow-unplayable` results in a garbage (but playable) stream. Also `--check-formats` doesn't work.

**Comment by Anuskuss** [comment]
Latest nightly (2023.08.14.182926) does not say "Maybe DRM" but `--allow-unplayable` results in a garbage (but playable) stream. But atleast `--check-formats` seems to be working now.

**Comment by pukkandan** [comment]
If you can attach dumps and logs, I can try to detect "maybe DRM" in `-F`. Nothing more can be done

**Comment by Anuskuss** [comment]
[dumps.zip](https://github.com/yt-dlp/yt-dlp/files/12389110/dumps.zip)

```
[debug] Command-line config: ['-vU', '--write-page', '-f292', 'https://www.youtube.com/watch?v=oAX5munikYU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481]
[debug] Python 3.11.3 (CPython x86_64 64bit) - Linux-6.4.11-arch2-1-x86_64-with-glibc2.38 (OpenSSL 3.1.2 1 Aug 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, brotlicffi-1.0.9.2, certifi-2023.07.22, sqlite3-2.6.0, websockets-10.4, xattr-0.10.1
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[youtube] Extracting URL: https://www.youtube.com/watch?v=oAX5munikYU
[youtube] oAX5munikYU: Downloading webpage
[youtube] Saving request to oAX5munikYU_https_-_consent.youtube.com_mcontinue=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DoAX5munikYU%26bpctr%3D9999999999%26has_verified%3D1%26cbrd%3D1_gl=DE_m=0_pc=yt_cm=4_hl=en_src=1.dump
WARNING: [youtube] unable to extract initial player response; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] oAX5munikYU: Downloading ios player API JSON
[youtube] Saving request to oAX5munikYU_https_-_www.youtube.com_youtubei_v1_playerkey=AIzaSyB-63vPrdThhKuerbB2N_l7Kwwcxj6yUAc_prettyPrint=false.dump
[youtube] oAX5munikYU: Downloading android player API JSON
[youtube] Saving request to oAX5munikYU_https_-_www.youtube.com_youtubei_v1_playerkey=AIzaSyA8eiZmM1FaDVjRy-df2KTyQ_vz_yYM39w_prettyPrint=false.dump
[youtube] oAX5munikYU: Downloading iframe API JS
[youtube] Saving request to oAX5munikYU_https_-_www.youtube.com_iframe_api.dump
[youtube] oAX5munikYU: Downloading player f980f2a9
[youtube] Saving request to oAX5munikYU_https_-_www.youtube.com_s_player_f980f2a9_player_ias.vflset_en_US_base.js.dump
[youtube] oAX5munikYU: Downloading web player API JSON
[youtube] Saving request to oAX5munikYU_https_-_www.youtube.com_youtubei_v1_playerkey=AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8_prettyPrint=false.dump
[youtube] Extracting URL: tnp_1rzDmro
[youtube] tnp_1rzDmro: Downloading webpage
[youtube] Saving request to tnp_1rzDmro_https_-_consent.youtube.com_mcontinue=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3Dtnp_1rzDmro%26bpctr%3D9999999999%26has_verified%3D1%26cbrd%3D1_gl=DE_m=0_pc=yt_cm=4_hl=en_src=1.dump
WARNING: [youtube] unable to extract initial player response; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] tnp_1rzDmro: Downloading ios player API JSON
[youtube] Saving request to tnp_1rzDmro_https_-_www.youtube.com_youtubei_v1_playerkey=AIzaSyB-63vPrdThhKuerbB2N_l7Kwwcxj6yUAc_prettyPrint=false.dump
[youtube] tnp_1rzDmro: Downloading android player API JSON
[youtube] Saving request to tnp_1rzDmro_https_-_www.youtube.com_youtubei_v1_playerkey=AIzaSyA8eiZmM1FaDVjRy-df2KTyQ_vz_yYM39w_prettyPrint=false.dump
[youtube] tnp_1rzDmro: Downloading iframe API JS
[youtube] Saving request to tnp_1rzDmro_https_-_www.youtube.com_iframe_api.dump
[youtube] tnp_1rzDmro: Downloading web player API JSON
[youtube] Saving request to tnp_1rzDmro_https_-_www.youtube.com_youtubei_v1_playerkey=AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8_prettyPrint=false.dump
[youtube] tnp_1rzDmro: Downloading m3u8 information
[youtube] Saving request to tnp_1rzDmro_https_-_manifest.googlevideo.com_api_manifest_hls_variant_expire_1692583735_ei_13LiZMaSBP-u6dsPtduBuA0_ip_2a02%3A8070%3A882%3Aada0%3A9bae%3A9eaf%3A8f28%3Ae550_id_b67a7fd6bcc39aba_source_youtube___b958314e30f5a3d6ee39a47d3e7bd3be.dump
WARNING: [youtube] unable to extract yt initial data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
WARNING: [youtube] Incomplete data received in embedded initial data; re-fetching using API.
[youtube] tnp_1rzDmro: Downloading initial data API JSON
[youtube] Saving request to tnp_1rzDmro_https_-_www.youtube.com_youtubei_v1_nextkey=AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8_prettyPrint=false.dump
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[info] tnp_1rzDmro: Downloading 1 format(s): 292
[debug] Invoking hlsnative downloader on "https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1692583735/ei/13LiZMaSBP-u6dsPtduBuA0/ip/2a02:8070:882:ada0:9bae:9eaf:8f28:e550/id/b67a7fd6bcc39aba/itag/292/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/gir/yes/clen/113512896/lmt/1439334338669851/dur/120.080/hls_chunk_host/rr2---sn-h0jeenld.googlevideo.com/mh/wf/mm/31,29/mn/sn-h0jeenld,sn-h0jeln7e/ms/au,rdu/mv/m/mvi/2/pl/40/initcwndbps/1718750/vprv/1/playlist_type/DVR/dover/13/mt/1692561666/fvip/5/short_key/1/keepalive/yes/fexp/24007246,51000024/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,gir,clen,lmt,dur,vprv,playlist_type/sig/AOq0QJ8wRgIhAKPK8QdAd-_HYAvVSiL7-mMx1JkByj6MhpMgvimPwUirAiEAyk3C4WyecTQ8lBHlVFFFMdY7Y0IigTL-TSc0PtVHvnA%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AG3C_xAwRgIhAIUu6fVlETQY9cs6vYH2q1D2A4E0XDZSZUtkDBqRM4yJAiEAtAAAWJn7o5kYNn1PorOH_Ue-kWGgCnHqYYZdqb5o9Lw%3D/playlist/index.m3u8"
[hlsnative] Downloading m3u8 manifest
ERROR: This format is DRM protected; Try selecting another format with --format or add --check-formats to automatically fallback to the next best format
```
